orchagent/orch.cpp

```diff
@@ -127,3 +206,97 @@ void Orch::dumpTuple(Consumer &consumer, KeyOpFieldsValuesTuple &tuple)
     }
     SWSS_LOG_DEBUG("%s\n", debug_msg.c_str());
 }
+
+ref_resolve_status Orch::resolveFieldRefArray(
+    type_map                    &type_maps,
+    const string                &field_name,
+    KeyOpFieldsValuesTuple      &tuple,
+    vector<sai_object_id_t>     &sai_object_arr)
+{
+    // example: [BUFFER_PROFILE_TABLE:e_port.profile0],[BUFFER_PROFILE_TABLE:e_port.profile1]
+    SWSS_LOG_ENTER();
+    size_t count = 0;
+    sai_object_arr.clear();
+    for (auto i = kfvFieldsValues(tuple).begin(); i != kfvFieldsValues(tuple).end(); i++)
+    {
+        if(fvField(*i) == field_name)
+        {
+            if(count > 1)
+            {
+                SWSS_LOG_ERROR("Singleton field with name:%s must have only 1 instance, actual count:%d\n", field_name.c_str(), count);
+                return ref_resolve_status::multiple_instances;
+            }
+            string ref_type_name, object_name;
+            string list = fvValue(*i);
+            vector<string> list_items;
+            if(list.find(list_item_delimiter) != string::npos)
+            {
+                list_items = tokenize(list, list_item_delimiter);
+            }
+            else
+            {
+                list_items.push_back(list);
+            }
+            for (size_t ind = 0; ind < list_items.size(); ind++)
+            {
+                if(!parseReference(type_maps, list_items[ind], ref_type_name, object_name))
+                {
+                    SWSS_LOG_ERROR("Failed to parse profile reference:%s\n", list_items[ind].c_str());
+                    return ref_resolve_status::not_resolved;
+                }
+                sai_object_id_t sai_obj = (*(type_maps[ref_type_name]))[object_name];
+                SWSS_LOG_DEBUG("Resolved to sai_object:0x%llx, type:%s, name:%s", sai_obj, ref_type_name.c_str(), object_name.c_str());
+                sai_object_arr.push_back(sai_obj);
+            }
+            count++;
+        }
+    }
+    if(0 == count)
+    {
+        SWSS_LOG_NOTICE("field with name:%s not found\n", field_name.c_str());
+        return ref_resolve_status::field_not_found;
+    }
+    return ref_resolve_status::success;
+}
+
+vector<string> Orch::parseNameArray(const string &input)
```

Comment:
Please remove this function, the tokenize() function supports single token case (find == string::npos)